src/PluginClient.php

```diff
@@ -79,7 +78,7 @@ public function sendRequest(RequestInterface $request)
         $pluginChain = $this->createPluginChain($this->plugins, function (RequestInterface $request) {
             try {
                 return new FulfilledPromise($this->client->sendRequest($request));
-            } catch (HttplugException $exception) {
+            } catch (\Exception $exception) {
```

Comment:
> How do we move forward? How do we avoid issues like this #33 to occur?

We need to take a decision of one of this three choices : 

 1. Do not change plugins / promise / httplug and add better documentation on the context of a promise in httplug
 2. Enforce this with a new interface and class like proposed in https://github.com/php-http/httplug/pull/116 (IMO, this is not a BC break but that's discutable)
 3. Make a BC break and allows everything as a rejected or resolved value and add checks everywhere where we use promise

In my opinion, 2 is the best solution as even if we add documentation, there are still people who doesn't read it and it would lead to the same error

However i'm ok with 1

3 is the worst thing to do IMO as we will increase all our libs with uneeded complexity.